modules/thermal_hydraulics/include/components/Component.h

```diff
@@ -223,6 +223,16 @@ class Component : public THMObject, public LoggingInterface, public NamingInterf
   void addDependency(const std::string & dependency);
 
 protected:
+  /**
+   * Gets a const reference to the THM problem
+   */
+  const THMProblem & getTHMProblem() const;
```

Comment:
So here you can overload, but not for mesh?

Reply:
Note that this particular overload is unnecessary/not useful, and I have since deleted it, since both `const THMProblem &` and `THMProblem &` versions can be `const` methods. In cases where an accessor method is returning a reference to data that actually belongs to the class, there should be the two different versions, but here, the member variable is a reference itself, but the data lives somewhere else.

As far as why the mesh accessor function cannot be overloaded: It *can* be overloaded, but when you overload, you lose the ability to be explicit about which version to use - the compiler decides based on the context. If the object is `const`, it chooses the `const` method; else, it chooses the other. This case is a bit special because the two versions are actually different - one makes a check, and the other doesn't.